questions.js

```diff
@@ -442,3 +442,208 @@ db.restaurants.find({
+    { borough: { $in: ["Manhattan", "Brooklyn"] } },
+    { cuisine: { $nin: ["American ", "Chinese"] } },
+  ],
+});
+
+// 43. Write a MongoDB query to find the restaurants that have a grade with a score of
+// 2 or a grade with a score of 6.
+
+db.restaurants.find({ "grades.score": { $in: [2, 6] } });
+
+// 44. Write a MongoDB query to find the restaurants that have a grade with a score
+// of 2 or a grade with a score of 6 and are located in the borough of Manhattan.
+
+db.restaurants.find({
+  $and: [{ "grades.score": { $in: [2, 6] } }, { borough: "Manhattan" }],
+});
+
+// 45. Write a MongoDB query to find the restaurants that have a grade with a score
+// of 2 or a grade with a score of 6 and are located in the borough of Manhattan or Brooklyn.
+
+db.restaurants.find({
+  $and: [
+    { "grades.score": { $in: [2, 6] } },
+    { borough: { $in: ["Manhattan", "Brooklyn"] } },
+  ],
+});
+
+// 46. Write a MongoDB query to find the restaurants that have a grade with a score
+// of 2 or a grade with a score of 6 and are located in the borough of Manhattan
+// or Brooklyn, and their cuisine is not American.
+
+db.restaurants.find({
+  $and: [
+    { "grade.score": { $in: [2, 6] } },
+    { Borough: { $in: ["Manhattan", "Brooklyn"] } },
+    { Cuisine: { $ne: "American " } },
+  ],
+});
+
+// 47. Write a MongoDB query to find the restaurants that have a grade with a
+// score of 2 or a grade with a score of 6 and are located in the borough of
+// Manhattan or Brooklyn, and their cuisine is not American or Chinese.
+
+db.restaurants.find({
+  $and: [
+    { "grade.score": { $in: [2, 6] } },
+    { Borough: { $in: ["Manhattan", "Brooklyn"] } },
+    { Cuisine: { $ne: "American ", $ne: "Chinese" } },
+  ],
+});
+
+// 48. Write a MongoDB query to find the restaurants that have all grades with a score greater than 5.
+
+db.restaurants.find({
+  grades: { $not: { $elemMatch: { score: { $lte: 5 } } } },
+});
+// &&
+db.restaurants.find({ "grades.score": { $not: { $lte: 5 } } });
+
+// 49. Write a MongoDB query to find the restaurants that have all grades with a
+// score greater than 5 and are located in the borough of Manhattan
+
+db.restaurants.find({
+  "grades.score": { $not: { $lte: 5 } },
+  borough: "Manhattan",
+});
+
+// 50. Write a MongoDB query to find the restaurants that have all grades
+// with a score greater than 5 and are located in the borough of Manhattan or Brooklyn.
+
+db.restaurants.find(
+  { "grades.score": { $not: { $lte: 5 } } },
+  { borough: { $in: ["Manhattan", "Brooklyn"] } }
+);
+
+// 51. Write a MongoDB query to find the average score for each restaurant.
+
+db.restaurants.aggregate([
+  {
+    $unwind: "$grades",
+  },
+  { $group: { _id: "$name", avgScore: { $avg: "$grades.score" } } },
+]);
+
+// 52. Write a MongoDB query to find the highest score for each restaurant.
+
+db.restaurants.aggregate([
+  {
+    $unwind: "$grades",
+  },
+  { $group: { _id: "$name", avgScore: { $max: "$grades.score" } } },
```

Comment:
in the group stage the variable name should be "maxScore" and not "avgScore"

Reply:
Got that, it was a silly mistake